### DISCUSSION TITLE
Proposal to allow for different views

### DISCUSSION THREAD

**Discussion by TimHal** [payloadcms/payload#1462] [Feature Requests & Ideas]
Following up on the discussion with 'digi' frmo the payload discord [here](https://discord.com/channels/967097582721572934/1043100541036789760), I want to propose the feature of conditional visibility of collections in the admin dashboard, regardless of a collection's read access.

**Use Case**
Given a fully public website which consists of a section for blog entries and news articles you'd probably want collections like `BlogEntries` and `NewsArticles` both with public read access. However, if there are different kinds of editors they might not be interested to see the other collection in their admin dashboard. The same is true for collections and globals that might be public but really only serve a technical purpose. 

**Proposal**
Allow users, depending on their role or some other property, to see a different subsection of collections, regardless of their access controls in an intuitive way.

**Implementation**
I can think of two ways to achieve this

1) Enrich the config for collections and globals by an `admin.isHidden` field of type `boolean | function` with a function of the same signature as `access.read`. The default value is `true` but in the function it is possible to access `req.user` and show/hide collections from the dashboard at runtime.

2) Implement a payload-wide `views` property in the root configuration. The main idea is to take a subset of the collections/globals and give them a name. This approach would be even more flexible as it could perhaps be possible to switch views at runtime - I attached a mockup for this below. 

![Screenshot from 2022-11-19 07-47-45](https://user-images.githubusercontent.com/12948708/202838751-c64df0cb-b8cf-4fcf-8e0f-927928f5a6b4.png)

**Current Solution**
I want to point out that it is possible to achieve these results right now by either (a) swapping in your own admin components, or (b) inspecting the query for read access and check if the `id` field is undefined (as gonza.moiguer kindly pointed out, thank you!). However, I would argue that these solutions do not add to payload's awesome DX and require quite some knowledge and work to implement - whereas the proposed solutions could make things quite fast and easy.

**Comment by niallobrien** [comment]
This would be an excellent feature to improve the UX for end-userrs.

**Comment by TimHal** [comment]
This proposal is implemented by https://github.com/payloadcms/payload/pull/2487
You can now conditionally hide collections or globals. Thanks to everyone for their feedback.
